template/server/main.py

```diff
@@ -113,6 +105,11 @@ async def post_execute(request: ExecutionRequest):
             status_code=404,
         )
 
+    # set global env vars if not set on first execution
+    if not ws.global_env_vars:
```

Comment:
if there are no global env vars you will call this for every execution, can we do it only once?